### PR TITLE
Add Compact to SQL doc [API-367]

### PR DESCRIPTION
**Pull request by sancar**
Also added compact to a couple of places that neeeds to be mentioned.
I have checked where Portable is mentioned and put Compact next to it.

### CODE REVIEW

File: docs/modules/cluster-performance/pages/best-practices.adoc [hunk 1]
```diff
@@ -1515,6 +1516,5 @@ Note that caching makes the query evaluation faster, but it consumes more memory
 * INDEX-ONLY: Deserialized values are cached only when they are inserted into an index.
 * ALWAYS: Deserialized values are always cached.
 
-If you are using portable serialization or your map's in-memory format is `OBJECT` or `NATIVE`,
-then `cache-deserialized-values` element does not have any effect.
+If you are using xref:serialization:implementing-portable-serialization.adoc[Portable] / xref:serialization:compact-serialization.adoc[Compact] serialization or your map's in-memory format is `OBJECT` or `NATIVE`, then `cache-deserialized-values` element does not have any effect.
```
Comment (mdumandag) [inline docs/modules/cluster-performance/pages/best-practices.adoc:1519]: Should we also make it [Compact(BETA)] to match with the other usages?
```suggestion
If you are using xref:serialization:implementing-portable-serialization.adoc[Portable] / xref:serialization:compact-serialization.adoc[Compact(BETA)] serialization or your map's in-memory format is `OBJECT` or `NATIVE`, then `cache-deserialized-values` element does not have any effect.
```

File: docs/modules/cluster-performance/pages/best-practices.adoc [hunk 2]
```diff
@@ -1507,6 +1507,7 @@ hazelcast:
       backup-count: 1
 ----
 ====
+--
```
Comment (mdumandag) [inline docs/modules/cluster-performance/pages/best-practices.adoc:1510]: Why is this necessary? I haven't seen it mentioned on https://github.com/hazelcast/hz-docs/blob/main/docs-tabs-library/tabs-block.js

File: docs/modules/data-structures/pages/creating-a-map.adoc [hunk 3]
```diff
@@ -231,6 +231,8 @@ You can load objects into a map. When you define a map that holds objects, you n
 
 * `Portable`: Provides partial deserialization for queries, improving query performance. Needs more memory to store metadata. See the xref:serialization:implementing-portable-serialization.adoc[Implementing Portable Serialization section].
 
+* `Compact(Beta)`: Provides partial deserialization for queries, improving query performance. More memory efficient than Portable. See the xref:serialization:compact-serialization.adoc[Implementing Compact Serialization section].
```
Comment (mdumandag) [inline docs/modules/data-structures/pages/creating-a-map.adoc:234]:
```suggestion
* `Compact(BETA)`: Provides partial deserialization for queries, improving query performance. More memory efficient than Portable. See the xref:serialization:compact-serialization.adoc[Implementing Compact Serialization section].
```
To be consistent with other places

File: docs/modules/performance/pages/caching-deserialized-values.adoc [hunk 4]
```diff
@@ -35,6 +35,7 @@ hazelcast:
       backup-count: 1
 ----
 ====
+--
```
Comment (mdumandag) [inline docs/modules/performance/pages/caching-deserialized-values.adoc:38]: Why is this necessary? I haven't seen it mentioned on https://github.com/hazelcast/hz-docs/blob/main/docs-tabs-library/tabs-block.js

File: docs/modules/performance/pages/caching-deserialized-values.adoc [hunk 5]
```diff
@@ -43,5 +44,5 @@ Note that caching makes the query evaluation faster, but it consumes more memory
 * INDEX-ONLY: Deserialized values are cached only when they are inserted into an index.
 * ALWAYS: Deserialized values are always cached.
 
-If you are using portable serialization or your map's in-memory format is `OBJECT` or `NATIVE`,
-then `cache-deserialized-values` element does not have any effect.
+If you are using xref:serialization:implementing-portable-serialization.adoc[Portable] / xref:serialization:compact-serialization.adoc[Compact] serialization or your map's in-memory format is `OBJECT` or `NATIVE`, then `cache-deserialized-values` element does not have any effect.
```
Comment (mdumandag) [inline docs/modules/performance/pages/caching-deserialized-values.adoc:47]:
```suggestion
If you are using xref:serialization:implementing-portable-serialization.adoc[Portable] / xref:serialization:compact-serialization.adoc[Compact(BETA)] serialization or your map's in-memory format is `OBJECT` or `NATIVE`, then `cache-deserialized-values` element does not have any effect.
```
Should we also make it [Compact(BETA)] to match with the other usages?

File: docs/modules/query/pages/querying-maps-sql.adoc [hunk 6]
```diff
@@ -78,7 +78,14 @@ The `this` alias is returned by a query only if the value does not contain an ob
 If you store objects in a map, you can query its fields only if the object is serialized using one of the following options:
 
 - For Java objects (xref:serialization:implementing-java-serializable.adoc[`Serializable`], xref:serialization:implementing-dataserializable.adoc[`DataSerializable`], xref:serialization:implementing-dataserializable.adoc#identifieddataserializable[`IdentifiedDataSerializable`]), only public fields and getters are available. A public field is available with its exact name. A getter is available with the JavaBean naming conventions.
-- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, the fields that are written in the `Portable.writePortable()` method are exposed using their exact names.
+- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, 
+	- the fields that are written in the `Portable.writePortable()` method are exposed using their exact names.
+	- the fields that are written in the `GenericRecordBuilder` API are exposed using their exact names. 
+- For xref:serialization:compact-serialization.adoc[`Compact`] objects,
```
Comment (mdumandag) [inline docs/modules/query/pages/querying-maps-sql.adoc:84]:
```suggestion
- For xref:serialization:compact-serialization.adoc[`Compact(BETA)`] objects, 
```

File: docs/modules/query/pages/querying-maps-sql.adoc [hunk 7]
```diff
@@ -78,7 +78,14 @@ The `this` alias is returned by a query only if the value does not contain an ob
...
-- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, the fields that are written in the `Portable.writePortable()` method are exposed using their exact names.
+- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, 
+	- the fields that are written in the `Portable.writePortable()` method are exposed using their exact names.
+	- the fields that are written in the `GenericRecordBuilder` API are exposed using their exact names. 
+- For xref:serialization:compact-serialization.adoc[`Compact`] objects, 
+	- the fields that are written in the `CompactSerializer.write()` method are exposed using their exact names.
+	- the fields that are written in the `GenericRecordBuilder` API are exposed using their exact names. 
+	- From members or from Java client, one can write `Compact` objects with zero-config. In that case, the fields will be exposed similar to Java Objects described above. Only public fields and getters are available. A public field is available with its exact name. A getter is available with the JavaBean naming conventions.
```
Comment (mdumandag) [inline docs/modules/query/pages/querying-maps-sql.adoc:87]: This nested list is not properly displayed on https://deploy-preview-151--hardcore-allen-f5257d.netlify.app/hazelcast/5.0-snapshot/query/querying-maps-sql.html#querying-object-fields

File: docs/modules/sql/pages/delete.adoc [hunk 8]
```diff
@@ -72,4 +72,4 @@ Whether you can access nested fields like this in objects depends on how your ma
 
 - For Java objects (xref:serialization:implementing-java-serializable.adoc[`Serializable`], xref:serialization:implementing-dataserializable.adoc[`DataSerializable`], xref:serialization:implementing-java-serializable.adoc#identifieddataserializable[`IdentifiedDataSerializable`]), you must xref:clusters:deploying-code-on-member.adoc[add the object's class to the classpath of the member] you are sending the query to.
 
-- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
+- For xref:serialization:implementing-portable-serialization.adoc[`Portable`]/xref:serialization:compact-serialization.adoc[`Compact(BETA)`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
```
Comment (mdumandag) [inline docs/modules/sql/pages/delete.adoc:75]:
```suggestion
- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] / xref:serialization:compact-serialization.adoc[`Compact(BETA)`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
```
To be consistent with other places

File: docs/modules/sql/pages/update.adoc [hunk 9]
```diff
@@ -75,4 +75,4 @@ Whether you can access nested fields like this in objects depends on how your ma
 
 - For Java objects (xref:serialization:implementing-java-serializable.adoc[`Serializable`], xref:serialization:implementing-dataserializable.adoc[`DataSerializable`], xref:serialization:implementing-java-serializable.adoc#identifieddataserializable[`IdentifiedDataSerializable`]), you must xref:clusters:deploying-code-on-member.adoc[add the object's class to the classpath of the member] you are sending the query to.
 
-- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
\ No newline at end of file
+- For xref:serialization:implementing-portable-serialization.adoc[`Portable`]/xref:serialization:compact-serialization.adoc[`Compact(BETA)`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
```
Comment (mdumandag) [inline docs/modules/sql/pages/update.adoc:78]:
```suggestion
- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] / xref:serialization:compact-serialization.adoc[`Compact(BETA)`] objects, you do not need to add the object's class to the classpath of the member, unless you want to use parameterized queries to compare one object against another.
```

**Review by mdumandag — COMMENTED** [review]
Looks good apart from minor comments

File: docs/modules/performance/pages/caching-deserialized-values.adoc (hunk 4 above)
Comment (sancar) [reply docs/modules/performance/pages/caching-deserialized-values.adoc:38]: I was looking for reference to understand how should it be. Thanks for the link. Looking at the link and other usages, it seems that we need to add two dashes before and after [source*] block.

**Review by mdumandag — APPROVED** [review]

File: docs/modules/sql/pages/imap-connector.adoc [hunk 10]
```diff
@@ -75,6 +76,41 @@ OPTIONS (
 
 For more information on `Portable` see xref:serialization:implementing-portable-serialization.adoc[].
 
+
+=== `Compact` Serialization(BETA)
+
+For this format, you need to specify additional options:
+
+* `keyCompactTypeName`, `valueCompactTypeName`
```
Comment (ihsandemir) [inline docs/modules/sql/pages/imap-connector.adoc:84]:
```suggestion
* `keyCompactTypeName` if the key is serialized as Compact
* `valueCompactTypeName` if the value is serialized as Compact
```

File: docs/modules/query/pages/querying-maps-sql.adoc [hunk 11]
```diff
@@ -77,9 +77,17 @@ The `this` alias is returned by a query only if the value does not contain an ob
...
-- For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, the fields that are written in the `Portable.writePortable()` method are exposed using their exact names.
-- `HazelcastJsonValue`
+* For Java objects (xref:serialization:implementing-java-serializable.adoc[`Serializable`], xref:serialization:implementing-dataserializable.adoc[`DataSerializable`], xref:serialization:implementing-dataserializable.adoc#identifieddataserializable[`IdentifiedDataSerializable`]), only public fields and getters are available. A public field is available with its exact name. A getter is available with the JavaBean naming conventions.
+* For xref:serialization:implementing-portable-serialization.adoc[`Portable`] objects, 
+** the fields that are written in the `Portable.writePortable()` method are exposed using their exact names.
+** the fields that are written in the `GenericRecordBuilder` API are exposed using their exact names. 
+* For xref:serialization:compact-serialization.adoc[`Compact(BETA)`] objects, 
+** the fields that are written in the `CompactSerializer.write()` method are exposed using their exact names.
```
[the remark this answers is not in the document]
Comment (sancar) [reply docs/modules/query/pages/querying-maps-sql.adoc:85]: Copy pasted from Portable. I think it is trying to say that whatever user is given used exactly as is. I will change bot Compact and Portable.

File: docs/modules/query/pages/querying-maps-sql.adoc (hunk 7 above)
Comment (sancar) [reply docs/modules/query/pages/querying-maps-sql.adoc]: Fixed it.

**Review by ihsandemir — APPROVED** [review]

**Comment by Serdaro** [comment]
Minor edits will be applied after the merge.
